### PR TITLE
Add prizes for vendor code shipping

### PR DESCRIPTION
**Pull request by k7hpn**
- Add the ability for a vendor code to apply a prize when a "Ship Date" is set
- Enhance vendor invoice import to bring in a branch id as well

### CODE REVIEW

File: src/GRA.Domain.Service/VendorCodeService.cs [hunk 1]
```diff
@@ -206,15 +212,17 @@ var jobDetails
...
+                        int detailsColumnId = 0;
                         int orderDateColumnId = 0;
                         int shipDateColumnId = 0;
-                        int detailsColumnId = 0;
                         var issues = new List<string>();
                         int row = 0;
                         int totalRows = 0;
                         int updated = 0;
                         int alreadyCurrent = 0;
+                        VendorCodeType vendorCodeType = null;
```
Comment (iafb) [inline src/GRA.Domain.Service/VendorCodeService.cs:225]: This value doesn't seem to ever be set?

File: src/GRA.Domain.Service/VendorCodeService.cs [hunk 2]
```diff
@@ -376,10 +391,41 @@ string shipDateString
                                                 issues.Add($"Issue reading details on row {row}: {ex.Message}");
                                             }
                                         }
+
+                                        if (excelReader.GetValue(branchColumnId) != null)
+                                        {
+                                            try
+                                            {
+                                                string branch
+                                                    = excelReader.GetString(branchColumnId);
```
Comment (iafb) [inline src/GRA.Domain.Service/VendorCodeService.cs:400]: I was getting errors on this saying "can't convert double to string", `excelReader.GetValue(branchColumnId).ToString()` seems to work well for getting the value as a string regardless of excel column type.

File: src/GRA.Domain.Service/VendorCodeService.cs (hunk 1 above)
Comment (k7hpn) [reply src/GRA.Domain.Service/VendorCodeService.cs:225]: Yeah, this was working but then I moved out functionality into a separate method. I've fixed it so it uses this now - it caches the VendorCodeType until the next loop saving a lookup if it's the same VendorCodeType (spoiler alert: it always is in our case).

File: src/GRA.Domain.Service/VendorCodeService.cs (hunk 2 above)
Comment (k7hpn) [reply src/GRA.Domain.Service/VendorCodeService.cs:400]: I tested this with a spreadsheet from the vendor and got no errors so I'm going to leave as-is.

**Pull request merged**